WordPress/Classes/Services/PostCoordinator+UploadActionUseCase.swift

```diff
@@ -0,0 +1,100 @@
+
+import Foundation
+
+extension PostCoordinator {
+    enum UploadAction {
+        /// Upload the post as is
+        ///
+        /// If the post was published locally, it will be published in the server.
+        case upload
+        /// Upload a revision in the server.
+        case remoteAutoSave
+        case nothing
+    }
+
+    /// Provides business logic for automatic uploads.
+    final class UploadActionUseCase {
```

Comment:
The issue I find with the term `UseCase` is that it means: "a specific situation in which a product or service could potentially be used", meaning that it's named after where we need to use this code from, rather than what this code is.  This'll work fine if we only ever use this code for the auto-upload use case, but will look weird if we ever want to use any of it to introspect posts for other reasons.

That said, I don't want to hold back this PR on my disagreement on the naming, so it's ok that you make a final call and move ahead.

Reply:
Thank you for the explanation. You're right. Now, I'm lost at what to name it. Because I do not agree with `*Logic`. I feel like it's at the same level as `*Utils` which gets abused. 😐 Anyway, in the interest of getting this PR merged to unblock the other pending issues, I used `*Interactor` for now. We can geek out on the name later in the life of the master branch. 😄